PHP migration guide

erikaheidi opened the pull request:
This PR introduces a PHP Migration guide with instructions covering **CLI workloads** and **Web Applications**. This is not supposed to be a traditional tutorial, but rather an entry point to direct users to more detailed resources. 

It covers also the handling of PHP extensions as it is now, and how to search for extensions.

This guide does not mention our Laravel image yet, because I am working first on a "getting started guide" and once that is published I will add a section here linking to that guide.

Preview: https://deploy-preview-1495--ornate-narwhal-088216.netlify.app/chainguard/migration/migrating-php/

erikaheidi committed:
first draft of PHP migration guide

erikaheidi committed:
small update

SharpRake commented on content/chainguard/migration-guides/migrating-to-php-images.md:
```diff
@@ -0,0 +1,187 @@
+## Migrating PHP Web applications to use Chainguard Images
+For PHP web applications that serve content through a web server, you should use the `latest-fpm` and `latest-fpm-dev` variants of our PHP image. Combine it with our [Nginx image](/chainguard/chainguard-images/reference/nginx) and an optional database for a traditional LEMP setup.
+
+The overall migration process is essentially the same as described in the previous section, with the difference that you won't set up application entry points, since these images run as services. Your Dockerfile may require additional steps to set up front end dependencies, initialize databases, and perform any additional tasks needed for the application to run through a web server.
```
```suggestion
The overall migration process is essentially the same as described in the previous section, with the difference that you won't set up application entry points, since these images run as services. Your Dockerfile may require additional steps to set up front-end dependencies, initialize databases, and perform any additional tasks needed for the application to run through a web server.
```

based mostly on [this](https://medium.com/@ist.stevkovski/is-it-front-end-or-front-end-or-frontend-3ae717cae4aa) which is not a reliable grammar reference but it looks right to me

SharpRake commented on content/chainguard/migration-guides/migrating-to-php-images.md:
```diff
@@ -0,0 +1,187 @@
+
+## Additional Resources
+
+Our [PHP image documentation](/chainguard/chainguard-images/reference/php/) covers details about all PHP image variants, including the list of available tags for both development and production images. For another example of LEMP setup using MariaDB, check our [Getting Started with the MariaDB Chainguard Image](https://edu.chainguard.dev/chainguard/chainguard-images/getting-started/mariadb/).
```
```suggestion
Our [PHP image documentation](/chainguard/chainguard-images/reference/php/) covers details about all PHP image variants, including the list of available tags for both development and production images. For another example of a LEMP setup using MariaDB, check our guide on [Getting Started with the MariaDB Chainguard Image](https://edu.chainguard.dev/chainguard/chainguard-images/getting-started/mariadb/).
```

SharpRake commented on content/chainguard/migration-guides/migrating-to-php-images.md:
```diff
@@ -0,0 +1,187 @@
+
+Our [PHP image documentation](/chainguard/chainguard-images/reference/php/) covers details about all PHP image variants, including the list of available tags for both development and production images. For another example of LEMP setup using MariaDB, check our [Getting Started with the MariaDB Chainguard Image](https://edu.chainguard.dev/chainguard/chainguard-images/getting-started/mariadb/).
+
+The [Debugging Distroless](/chainguard/chainguard-images/debugging-distroless-images/) guide contains important information for debugging issues with distroless images. Check also the [Verifying Images](/chainguard/chainguard-images/verifying-chainguard-images-and-metadata-signatures-with-cosign/) resource for details around provenance, SBOMs, and image signatures.
```
```suggestion
The [Debugging Distroless](/chainguard/chainguard-images/debugging-distroless-images/) guide contains important information for debugging issues with distroless images. You can also refer to the [Verifying Images](/chainguard/chainguard-images/verifying-chainguard-images-and-metadata-signatures-with-cosign/) resource for details around provenance, SBOMs, and image signatures.
```

"check also" is kind of a funny way to start a sentence

SharpRake commented on content/chainguard/migration-guides/migrating-to-php-images.md:
```diff
@@ -0,0 +1,187 @@
+---
+
+Chainguard Images are built on top of [Wolfi](/chainguard/open-source/wolfi/), a Linux _undistro_ designed specifically for containers. Our PHP images have a minimal design that ensures a smaller attack surface, which results in smaller images [with low to zero](/chainguard/chainguard-images/vuln-comparison/php/) CVEs. Nightly builds deliver fresh images whenever updated packages are available, w ...
+
```
One general note: there isn't a lot of consistency in whether it's stylized as "Image" or "image". I think for any instance where you're talking about one or more Chainguard images, you should capitalize the "I", and in cases where you're talking about images generally, it's fine to just leave it lowercase.

erikaheidi replied:
Do you mean for instance "Our PHP Images" vs "Our PHP images"  ? The way I have been using it so far is whenever it's together "Chainguard Images" I make sure is uppercase, because it's the product name. But if I'm referring to the unit of image (one or more, seen as objects) then I am using the lowercase.

SharpRake reviewed (approved):
This reads great! I left some suggestions, mostly around grammar stuff.

One thing, i wasn't able to get through the "Getting started with PHP" guide. I think it may have just been my setup (I didn't have Composer installed nor a PHP environment to work with) but I kept running into errors when I tried the first docker command in [this section](https://deploy-preview-1495--ornate-narwhal-088216.netlify.app/chainguard/chainguard-images/getting-started/php/#step-1-setting-up-a-demo-application).

erikaheidi commented on content/chainguard/migration-guides/migrating-to-php-images.md:
```diff
@@ -0,0 +1,187 @@
+
+Chainguard PHP Images offer multiple variants catering to distinct use cases, such as command-line interfaces (CLI) and web applications. Each variant comes in two flavors: a minimal runtime image (distroless) and a builder image distinguished by the `-dev` suffix (e.g., `latest-dev`).
+
+In a nutshell, distroless images don't include a package manager or a shell, being used exclusively as runtimes to keep the environment to a minimum. Builder images, on the other hand, include packages such as `apk` and `composer` for building PHP applications. Builder images can be used _as-is_ to provide a more straightforward migration path, but we encourage users to combine both images in a m ...
```
yeah that one I am always unsure! will stick to the docker convention then

erikaheidi committed:
Apply suggestions from code review

Co-authored-by: Mark Drake <33191761+SharpRake@users.noreply.github.com>
Signed-off-by: Erika Heidi <erikaheidi@users.noreply.github.com>

erikaheidi committed:
Updating php getting started guide to fix composer command

erikaheidi commented:
> This reads great! I left some suggestions, mostly around grammar stuff.
> 
> One thing, i wasn't able to get through the "Getting started with PHP" guide. I think it may have just been my setup (I didn't have Composer installed nor a PHP environment to work with) but I kept running into errors when I tried the first docker command in [this section](https://deploy-preview-1495--ornate-narwhal-088216.netlify.app/chainguard/chainguard-images/getting-started/php/#step-1-setting-up-a-demo-application).

Thanks Mark! Also, good catch with the getting started guide. The `-dev` image changed a bit and now runs as a regular user (I think it run as root in the beginning) so I included an update to that guide as well.

erikaheidi committed:
reorganizing migration menu

erikaheidi commented:
I made some changes to the menu "migration guides" based on what we discussed in a recent meeting talking about this topic. Renamed to `migration` and added a redirect rule; added a folder for the specific migration guides resembling what we have for the "getting started" guides. This is what it looks like after the changes:

![Screenshot from 2024-04-05 16-17-01](https://github.com/chainguard-dev/edu/assets/293241/cc5774e3-2f8b-47a6-9388-8c728dfd61e3)

erikaheidi committed:
Reorganizing menu
